### PR TITLE
Feat/database setup

### PR DESCRIPTION
**Pull request by rossman-andrew**
https://trello.com/c/D6na0zEz/22-set-up-mongo-database-to-correctly-store-user-click-events
https://trello.com/c/4PpqqPIM/23-write-script-for-generating-user-click-event-data
https://trello.com/c/cAz0Uoe0/25-set-up-mongo-database-to-retrieve-all-events-for-a-single-users-completed-session
https://trello.com/c/sP94qhu3/10-set-up-basic-server-for-events

### CODE REVIEW

File: database/dummyData/userEventData.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    profile[value] = profileValues[index];
+  });
+  return profile;
+};
+
+const generateMovieObj = () => {
+  const movieObj = {
+    id: Math.floor(Math.random() * 300000),
+    profile: generateMovieProfile(),
+    isRec: Math.floor(Math.random() * 2),
```
Comment (shwetachari) [inline database/dummyData/userEventData.js]: If you are generating a value that is either 0 or 1, then you can shorten this to Math.round(Math.random()). Also, in our App plan, the expected input for this field is a boolean, so you might want to add a !!

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    Event.findOne({ _id: session }, (err, item) => {
+      if (err) {
+        callback(err, null);
+      } else {
+        callback(null, item);
+      }
+    });
+  });
+
+  const addEvent = ((session, queryObj) => {
```
Comment (shwetachari) [inline database/index.js]: Is there a reason you're using callbacks instead of promises? This doesn't seem heavily nested or anything, so it's not a problem here, but I can see things getting complicated in the future.

File: server/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  db.selectAllEvents((err, data) => {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.status(200).send(data);
+    }
+  });
+});
+
+app.get('/userEventData', (req, res) => {
```
Comment (shwetachari) [inline server/index.js]: Is there a reason this is a get request instead of a post request? It seems like this is generating data to put into your database (post-like) as opposed to querying your database to return data with the response (get-like).

**Review by shwetachari — COMMENTED** [review]
Overall, it looks good! I just had some minor concerns (that you can choose to address at your discretion).
1. The isRec field in the received movie object may need to be changed from a number to a boolean
2. Nested callbacks (stylistic concern, not too important)
3. Express routing (semantic misunderstanding)

File: database/dummyData/userEventData.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  };
+  const profileKeys = Object.keys(profile);
+  profileKeys.forEach((value, index) => {
+    profile[value] = profileValues[index];
+  });
+  return profile;
+};
+
+const generateMovieObj = () => {
+  const rec = Math.round(Math.random()) === 0;
```
Comment (claracho) [inline database/dummyData/userEventData.js:95]: Cool way of converting random 0/1 to boolean!

File: database/dummyData/userEventData.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    profileValues.push(genreVal);
+  }
+  const genreRatio = 100 / genreSum;
+  let newGenreSum = 0;
+  profileValues.forEach((value, index) => {
+    const newGenreVal = Math.floor(profileValues[index] * genreRatio);
+    newGenreSum += newGenreVal;
+    profileValues[index] = newGenreVal;
+  });
+  profileValues.push(100 - newGenreSum);
```
Comment (claracho) [inline database/dummyData/userEventData.js:68]: Will this result in 16 elements for profileValues?

**Review by claracho — APPROVED** [review]
Looks good!
